### ISSUE TITLE
Keyboard handling events doesn't work on custom day templates 

### ISSUE THREAD

**Issue by MadMax888** [ng-bootstrap/ng-bootstrap#2946]
### Bug description:
The current datepicker doesn't support keyboard event inside custom day template

### Link to minimally-working StackBlitz that reproduces the issue:

https://stackblitz.com/edit/angular-cflf4j

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 7.0.0

ng-bootstrap: 4.0.1

Bootstrap:

**Comment by maxokorokov** [comment]
That is expected, because the element in question is never focused. 
Focus is received only by the wrapping `<div class="ngb-dp-day"> ... </div>` and currently all the key events are handled by the private `NgbDatepickerKeymapService` → https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/datepicker/datepicker-keymap-service.ts#L23

Could you please elaborate on what you're trying to achieve and we'll see what could be done.

**Comment by MadMax888** [comment]
@maxokorokov Yep, due to this, basic keyboard actions don't work in custom templates. And then you have to do smth like this `<ng-template
  #rangeDay
  let-date="date"
  let-disabled="disabled"
  let-selected="selected"
  let-currentMonth="currentMonth">
  <span
    class="custom-day text-center"
    [ngClass]="{...}"
    (document:keyup)="onKeyPress($event, currentMonth, $event.target.innerText, date)">
    {{ date.day }}
  </span>
</ng-template>`

`(document:keyup)="onKeyPress($event, currentMonth, $event.target.innerText, date)`
inside function ` ... if (Number(curDay) === date.day && Number(curMonth) === date.month && Key[evt.which]) {
      switch (evt.which) { ...` - this very expensive operation for performance, no convenient , and looks like hack.
